mne/viz/_3d.py

```diff
@@ -1344,3 +1345,135 @@ def _get_view_to_display_matrix(scene):
+    subject : str
+        The subject name corresponding to FreeSurfer environment variable
+        SUBJECT.
+    subjects_dir : None | str
+        The path to the freesurfer subjects reconstructions. It corresponds to
+        Freesurfer environment variable SUBJECTS_DIR. The default is None.
+    scale_factor : float
+        The scaling applied to convert amplitudes to voxel sizes for the plot.
+        Defaults to 1e9.
+    block : bool
+        Whether to halt program execution until the figure is closed. Defaults
+        to False.
+    show : bool
+        Show figure if True. Defaults to True.
+
+    Returns
+    -------
+    fig : instance of matplotlib figure
+        The figure containing 3-D locations of the dipoles.
+    """
+    import matplotlib.pyplot as plt
+    from mpl_toolkits.mplot3d import Axes3D  # noqa
+    try:
+        import nibabel as nib
+    except ImportError:
+        raise
+    subjects_dir = get_subjects_dir(subjects_dir=subjects_dir,
+                                    raise_error=True)
+    t1_fname = op.join(subjects_dir, subject, 'mri', 'T1.mgz')
+    t1 = nib.load(t1_fname)
+    ras2vox = t1.header.get_ras2vox()
+    data = t1.get_data()
+    dims = len(data)  # Symmetric size assumed.
+    gridx, gridy = np.meshgrid(np.linspace(0, dims, dims),
+                               np.linspace(0, dims, dims))
+
+    fig = plt.figure()
+    ax = fig.add_subplot(111, projection='3d')
+
+    trans = _get_trans(trans, fro='head', to='mri')[0]
+    points = [apply_trans(trans['trans'], loc) for loc in dipole.pos]
+    ori = [apply_trans(np.linalg.inv(trans['trans']), o) for o in dipole.ori]
+
+    # Position in meters -> voxels.
+    points = np.array([apply_trans(ras2vox, loc * 1000.) for loc in points])
+
+    _plot_dipole(ax, data, points, 0, dipole, gridx, gridy, ori, scale_factor)
+    params = {'ax': ax, 'data': data, 'idx': 0, 'dipole': dipole,
+              'points': points, 'gridx': gridx, 'gridy': gridy, 'ori': ori,
+              'scale_factor': scale_factor}
+    ax.view_init(elev=30, azim=-140)
+
+    callback_func = partial(_dipole_changed, params=params)
+    fig.canvas.mpl_connect('scroll_event', callback_func)
+    fig.canvas.mpl_connect('key_press_event', callback_func)
+
+    plt_show(show, block=block)
+    return fig
+
+
+def _plot_dipole(ax, data, points, idx, dipole, gridx, gridy, ori,
+                 scale_factor):
+    """Plot dipoles."""
+    import matplotlib.pyplot as plt
+    # Plot slices.
+    xidx = int(round(points[idx, 0]))
+    yidx = int(round(points[idx, 1]))
+    zidx = int(round(points[idx, 2]))
+    ax.contourf(data[xidx][::-1], gridx, gridy, offset=0, zdir='x',
+                cmap='gray', zorder=0, alpha=0.5)
+    ax.contourf(gridx, gridy, data[:, yidx][::-1].T, offset=0, zdir='z',
+                cmap='gray', zorder=0, alpha=0.5)
+    ax.contourf(gridx, data[:, :, zidx][::-1].T[::-1], gridy, offset=0,
+                zdir='y', cmap='gray', zorder=0, alpha=0.5)
```

Comment:
It could be as easy as using [nibabel.processing.resample_from_to](http://nipy.org/nibabel/reference/nibabel.processing.html#nibabel.processing.resample_from_to), with the `affine` being our `mri_head_t['trans']`. Then you can just take `ndarray` slices of the resulting resampled image, instead of the original.